### PR TITLE
Cache some denotation fields in Symbols

### PR DESCRIPTION
**Pull request by odersky**

### CODE REVIEW

File: compiler/src/dotty/tools/dotc/core/Symbols.scala [hunk 1]
```diff
@@ -91,10 +91,24 @@ object Symbols {
 
     private[core] def invalidateDenotCache(): Unit = { checkedPeriod = Nowhere }
 
+    private[core] var cachedInfoOrCompleter: Type = _
+    private[core] var cachedFlags: FlagSet = _
+    private[core] var cachedName: Name = _
```
Comment (smarter) [inline compiler/src/dotty/tools/dotc/core/Symbols.scala:96]: It's minor, but these fields will appear after the `val id: Int` field in the object layout, meaning that there's a higher chance that calling them will require an extra memory load.

**Comment by odersky** [comment]
Test performance please

**Comment by odersky** [comment]
test performance please

**Comment by odersky** [comment]
test performance please

**Comment by odersky** [comment]
Does not seem to help much. Closing,.

**Pull request closed without merging**